kubectl-fdb/cmd/fix_coordinator_ips_test.go

```diff
@@ -0,0 +1,233 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[plugin] fix-coordinator-ips command", func() {
+	When("building cluster file update commands", func() {
+		clusterName := "test"
+		namespace := "test"
+
+		var cluster fdbtypes.FoundationDBCluster
+		var podList corev1.PodList
+
+		type testCase struct {
+			Context          string
+			ExpectedCommands [][]string
+			ExpectedError    string
+		}
+
+		BeforeEach(func() {
+			cluster = fdbtypes.FoundationDBCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterName,
+					Namespace: namespace,
+				},
+				Spec: fdbtypes.FoundationDBClusterSpec{
+					ProcessCounts: fdbtypes.ProcessCounts{
+						Storage: 1,
+					},
+				},
+			}
+
+			podList = corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "instance-1",
+							Namespace: namespace,
+							Labels: map[string]string{
+								fdbtypes.FDBProcessClassLabel: string(fdbtypes.ProcessClassStorage),
+								fdbtypes.FDBClusterLabel:      clusterName,
+							},
+						},
+					},
+				},
+			}
+		})
+
+		DescribeTable("should execute the provided command",
+			func(input testCase) {
+				scheme := runtime.NewScheme()
+				_ = clientgoscheme.AddToScheme(scheme)
+				_ = fdbtypes.AddToScheme(scheme)
+				kubeClient := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(&cluster, &podList).Build()
+				cluster.Status.ConnectionString = "test:test@127.0.0.1:4501"
+
+				commands, err := buildClusterFileUpdateCommands(&cluster, kubeClient, input.Context, namespace)
+
+				if input.ExpectedError != "" {
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(Equal(input.ExpectedError))
+				} else {
+					Expect(err).NotTo(HaveOccurred())
+					Expect(commands).To(HaveLen(len(input.ExpectedCommands)))
+					for index, command := range commands {
+						Expect(command.Args).To(Equal(input.ExpectedCommands[index]))
+					}
+				}
+			},
+			Entry("instance with valid pod",
+				testCase{
+					ExpectedCommands: [][]string{
+						{
+							"/usr/local/bin/kubectl",
```

Comment:
My only concern here would be that the test cases assume that `kubectl` is installed under `/usr/local/bin/kubectl` if not the tests will be failing with a different value. We could somehow inject the path for testing only, that would allow these tests to be run without having `kubectl` in that location (or installed at all).